python/ray/train/xgboost/xgboost_predictor.py

```diff
@@ -115,7 +115,9 @@ def _predict_pandas(
             if feature_columns:
                 data = data[:, feature_columns]
         elif feature_columns:
-            data = data[feature_columns]
+            data = data[feature_columns].to_numpy()
```

Comment:
Shouldn't a pandas dataframe be compatible with Xgboost dmatrix already?

Reply:
It's not because the dtype is TensorType, which is not supported by DMatrix. Btw, this was only caught because we introduced these tests ;-)